lib/bup/drecurse.py

```diff
@@ -37,15 +45,37 @@ def stat(self):
         return xstat.fstat(self.fd)
 
 
+def _filename_stat_generator(path):
+    """Yields all file entry names (not prefixed with path) and their
+    stat info; stat errors are skipped over with add_error().
+
+    Uses a fast/streaming, scandir based traversal when scandir is available.
+    """
+    if scandir is not None:
+        for entry in scandir.scandir(path):
+            n = entry.name
+            try:
+                # We need to use xstat instead of entry.stat() because
+                # entry.stat() uses Python's normal floating-point mtimes.
+                st = xstat.lstat(n)
+                yield (n,st)
+            except OSError as e:
+                add_error(Exception('%s: %s' % (resolve_parent(n), str(e))))
```

Comment:
this is strange - you copied it from the one place that was already weird, with the Exception() think? normally add_error just takes a string